### PR TITLE
Add Welcome pages for DCRDEX and CEX

### LINKED ISSUE #191: [DCRDEX] Create welcome page
use figma!

Dex Onboarding

### LINKED ISSUE #192: [CEX]: create welcome page
use figma!

Cex Onboarding

### PR DESCRIPTION
**Pull request by Vanvians**
Closes #191 
Closes #192 

Adds welcome page on first visit of DCRDEX page
Adds welcome page to Centralized Exchange page
Screenshot 2023-11-01 190022
Screenshot 2023-11-01 190231

### CODE REVIEW

File: ui/page/dcrdex/dcrdex_page.go [hunk 1]
```diff
@@ -7,6 +7,8 @@ import (
 	"github.com/crypto-power/cryptopower/app"
 	"github.com/crypto-power/cryptopower/ui/cryptomaterial"
 	"github.com/crypto-power/cryptopower/ui/load"
+	"github.com/crypto-power/cryptopower/ui/page/components"
+	"github.com/crypto-power/cryptopower/ui/values"
 )
 
 const DCRDEXID = "DCRDEXID"
```
Comment (itswisdomagain) [inline ui/page/dcrdex/dcrdex_page.go:14]: Not added by your PR, but `DCRDEXPageID`? I don't know what `DCRDEXID` mean.

File: ui/page/dcrdex/dex_splash_page.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+	"github.com/crypto-power/cryptopower/ui/cryptomaterial"
+	"github.com/crypto-power/cryptopower/ui/modal"
+	"github.com/crypto-power/cryptopower/ui/page/components"
+	"github.com/crypto-power/cryptopower/ui/values"
+)
+
+func (pg *DEXPage) initSplashPageWidgets() {
+	_, pg.splashPageInfoButton = components.SubpageHeaderButtons(pg.Load)
+	pg.enableDEXBtn = pg.Theme.Button(values.String(values.StrBack))
```
Comment (itswisdomagain) [inline ui/page/dcrdex/dex_splash_page.go:16]: enableDEXBtn uses strBack as the text??

File: ui/page/dcrdex/dcrdex_page.go [hunk 3]
```diff
@@ -34,6 +39,9 @@ func NewDEXPage(l *load.Load) *DEXPage {
 		MasterPage:        app.NewMasterPage(DCRDEXID),
 		openTradeMainPage: l.Theme.NewClickable(false),
 	}
+
+	dp.initSplashPageWidgets()
```
Comment (itswisdomagain) [inline ui/page/dcrdex/dcrdex_page.go:43]: This method only does 2 things. Doesn't need to be a method.

File: ui/page/dcrdex/dcrdex_page.go [hunk 4]
```diff
@@ -63,6 +70,9 @@ func (pg *DEXPage) OnNavigatedTo() {
 // eventually drawn on screen.
 // Part of the load.Page interface.
 func (pg *DEXPage) Layout(gtx C) D {
+	if !pg.WL.AssetsManager.IsDexFirstVisit() {
+		return components.UniformPadding(gtx, pg.splashPage)
+	}
```
Comment (itswisdomagain) [inline ui/page/dcrdex/dcrdex_page.go:75]: Uhm, you show the splash page if it is not the first visit?? Shouldn't this be reversed?

Comment (itswisdomagain) [inline ui/page/dcrdex/dcrdex_page.go:75]: Uhm, you show the splash page if it is not the first visit? Should this be reversed?

File: ui/page/dcrdex/dcrdex_page.go [hunk 5]
```diff
@@ -87,6 +97,12 @@ func (pg *DEXPage) HandleUserInteractions() {
 	if pg.CurrentPage() != nil {
 		pg.CurrentPage().HandleUserInteractions()
 	}
+	if pg.splashPageInfoButton.Button.Clicked() {
+		pg.showInfoModal()
+	}
+	if pg.navigateToSettingsBtn.Button.Clicked() {
+		pg.WL.AssetsManager.SetDexFirstVisit(true)
+	}
```
Comment (itswisdomagain) [inline ui/page/dcrdex/dcrdex_page.go:105]: I've seen this `navigateToSettingsBtn` a couple times and still don't know what it is. And looking at this click handler, it doesn't navigate to anywhere! Please rename the button to something that better describes what the button is meant to do.

File: ui/page/dcrdex/dcrdex_page.go [hunk 6]
```diff
@@ -34,6 +39,9 @@ func NewDEXPage(l *load.Load) *DEXPage {
 		MasterPage:        app.NewMasterPage(DCRDEXID),
 		openTradeMainPage: l.Theme.NewClickable(false),
 	}
+
+	dp.initSplashPageWidgets()
+	dp.navigateToSettingsBtn = dp.Theme.Button(values.String(values.StrStartTrading))
```
Comment (itswisdomagain) [inline ui/page/dcrdex/dcrdex_page.go:44]: Oh, I see the text on the button now. Please rename `navigateToSettingsBtn` to `finalizeOnboardingBtn`. To confirm, this is the button that is used to exit the onboarding view, correct?

File: ui/page/dcrdex/dcrdex_page.go [hunk 7]
```diff
@@ -87,6 +97,12 @@ func (pg *DEXPage) HandleUserInteractions() {
 	if pg.CurrentPage() != nil {
 		pg.CurrentPage().HandleUserInteractions()
 	}
+	if pg.splashPageInfoButton.Button.Clicked() {
+		pg.showInfoModal()
+	}
+	if pg.navigateToSettingsBtn.Button.Clicked() {
+		pg.WL.AssetsManager.SetDexFirstVisit(true)
```
Comment (itswisdomagain) [inline ui/page/dcrdex/dcrdex_page.go:104]: 2 things.
1) Don't save to db. Just toggle a variable on this page for now.
2) After toggling, reload the window to update the display.

File: libwallet/assets_config.go [hunk 8]
```diff
@@ -278,3 +278,15 @@ func (mgr *AssetsManager) SetTotalBalanceVisibility(data bool) {
...
+func (mgr *AssetsManager) IsDexFirstVisit() bool {
+	var data bool
+	mgr.db.ReadWalletConfigValue(sharedW.DexFirstVisitKey, &data)
+	return data
+}
+
+// SetDexFirstVisit sets the first visit to DCRDEX .
+func (mgr *AssetsManager) SetDexFirstVisit(data bool) {
+	mgr.db.SaveWalletConfigValue(sharedW.DexFirstVisitKey, data)
+}
```
Comment (itswisdomagain) [inline libwallet/assets_config.go:292]: Let's remove these config stuffs. We won't be needing them. Instead, a future PR will instantiate the DEX client core object and check if the user has set a DEX trading password. Until the user sets a DEX trading password, we'll keep showing the onboarding page.

**Review by itswisdomagain — CHANGES_REQUESTED** [review]
First pass.

File: ui/page/dcrdex/dcrdex_page.go (hunk 1 above)
Comment (Vanvians) [reply ui/page/dcrdex/dcrdex_page.go:14]: Changed to DCRDEXPageID

File: ui/page/dcrdex/dcrdex_page.go (hunk 3 above)
Comment (Vanvians) [reply ui/page/dcrdex/dcrdex_page.go:43]: Method is used by trade page to display DCRDEX page

**Review by ukane-philemon — APPROVED** [review]
LGTM.

I have no idea about how the welcome pages should look on full screen but my guts tell me the decision btn should be close to the text and top and bottom should have equal   space.

Screenshot 2023-12-05 at 2 02 48 PM

File: ui/cryptomaterial/icon_gallery.go [hunk 9]
```diff
@@ -28,7 +28,7 @@ type Icons struct {
 	RevealIcon, InfoAction, LightMode, DarkMode, AddIcon, ChevronRight, AddExchange, FlypMeIcon, ChangellyIcon,
 	SimpleSwapIcon, SwapzoneIcon, ShapeShiftIcon, GodexIcon, CoinSwitchIcon, ChangeNowIcon, CaretUp, CaretDown,
 	LTCBackground, LTCGroupIcon, DCRBackground, DCRGroupIcon, BTCBackground, BTCGroupIcon, CrossPlatformIcon,
-	IntegratedExchangeIcon, MultiWalletIcon, Dot *Image
+	IntegratedExchangeIcon, MultiWalletIcon, Dot, TradeExchange *Image
```
Comment (JustinBeBoy) [inline ui/cryptomaterial/icon_gallery.go:31]:
```suggestion
	IntegratedExchangeIcon, MultiWalletIcon, Dot, TradeExchangeIcon *Image
```

File: ui/cryptomaterial/icon_gallery.go [hunk 10]
```diff
@@ -133,6 +133,7 @@ func (i *Icons) DefaultIcons() *Icons {
 	i.ConcealIcon = NewImage(decredIcons["reveal"])
 	i.RevealIcon = NewImage(decredIcons["hide"])
 	i.AddExchange = NewImage(decredIcons["add_exchange"])
+	i.TradeExchange = NewImage(decredIcons["ri_exchange-line"])
```
Comment (JustinBeBoy) [inline ui/cryptomaterial/icon_gallery.go:136]: please change file name from `ri_exchange-line` -> ` trade-exchange-Icon`

File: ui/page/dcrdex/dex_splash_page.go [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+	"github.com/crypto-power/cryptopower/ui/modal"
+	"github.com/crypto-power/cryptopower/ui/page/components"
+	"github.com/crypto-power/cryptopower/ui/values"
+)
+
+func (pg *DEXPage) initSplashPageWidgets() {
+	_, pg.splashPageInfoButton = components.SubpageHeaderButtons(pg.Load)
+}
+
+func (pg *DEXPage) splashPage(gtx layout.Context) layout.Dimensions {
```
Comment (JustinBeBoy) [inline ui/page/dcrdex/dex_splash_page.go:18]:
```suggestion
func (pg *DEXPage) splashPage(gtx C) D {
```

File: ui/page/dcrdex/dex_splash_page.go [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+							pgContent.Alignment = text.Middle
+							return layout.Inset{Top: values.MarginPadding10}.Layout(gtx, pgContent.Layout)
+						}),
+					)
+				}),
+				layout.Stacked(pg.splashPageInfoButton.Layout),
+			)
+		}),
+		layout.Rigid(func(gtx C) D {
+			gtx.Constraints.Min.X = gtx.Dp(values.MarginPadding350)
```
Comment (JustinBeBoy) [inline ui/page/dcrdex/dex_splash_page.go:56]: why is 350?

File: ui/page/exchange/cex_splash_page.go [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+	"github.com/crypto-power/cryptopower/ui/page/components"
+	"github.com/crypto-power/cryptopower/ui/values"
+)
+
+func (pg *CreateOrderPage) initSplashPageWidgets() {
+	_, pg.splashPageInfoButton = components.SubpageHeaderButtons(pg.Load)
+	pg.enableDEXBtn = pg.Theme.Button(values.String(values.StrBack))
+}
+
+func (pg *CreateOrderPage) splashPage(gtx layout.Context) layout.Dimensions {
```
Comment (JustinBeBoy) [inline ui/page/exchange/cex_splash_page.go:19]:
```suggestion
func (pg *CreateOrderPage) splashPage(gtx C) D {
```

File: ui/cryptomaterial/icon_gallery.go (hunk 10 above)
Comment (dreacot) [reply ui/cryptomaterial/icon_gallery.go:136]: `trade_exchange_icon`

**Comment by ukane-philemon** [comment]
Figma: 
Screenshot 2023-12-05 at 7 10 43 PM
Screenshot 2023-12-05 at 7 11 02 PM

This PR after changes in https://github.com/crypto-power/cryptopower/pull/224/commits/0ba6c10cdd185884a8c25a1fc9b987f1e0199411

Screenshot 2023-12-05 at 7 23 02 PM
Screenshot 2023-12-05 at 7 23 09 PM

File: ui/page/dcrdex/dex_splash_page.go (hunk 12 above)
Comment (ukane-philemon) [reply ui/page/dcrdex/dex_splash_page.go]: removed.

File: ui/page/exchange/cex_splash_page.go (hunk 13 above)
Comment (ukane-philemon) [reply ui/page/exchange/cex_splash_page.go]: resolved.

File: ui/page/dcrdex/dex_splash_page.go (hunk 11 above)
Comment (ukane-philemon) [reply ui/page/dcrdex/dex_splash_page.go]: resolved.

File: ui/cryptomaterial/icon_gallery.go (hunk 10 above)
Comment (ukane-philemon) [reply ui/cryptomaterial/icon_gallery.go]: resolved.

File: ui/cryptomaterial/icon_gallery.go (hunk 9 above)
Comment (ukane-philemon) [reply ui/cryptomaterial/icon_gallery.go]: resolved.

File: ui/page/exchange/cex_splash_page.go [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+				Right: values.MarginPadding16,
+			}.Layout(gtx, pg.navToSettingsBtn.Layout)
+		}),
+	)
+}
+
+func (pg *CreateOrderPage) showInfoModal() {
+	info := modal.NewCustomModal(pg.Load).
+		Title(values.String(values.StrCentralizedExchangeCex)).
+		Body(values.String(values.StrDexInfo)).
```
[the remark this answers is not in the document]
Comment (ukane-philemon) [reply ui/page/exchange/cex_splash_page.go]: resolved.

File: ui/page/dcrdex/dcrdex_page.go [hunk 15]
```diff
@@ -24,24 +26,30 @@ type DEXPage struct {
...
 	dp := &DEXPage{
 		Load:              l,
-		MasterPage:        app.NewMasterPage(DCRDEXID),
+		MasterPage:        app.NewMasterPage(DCRDEXPageID),
 		openTradeMainPage: l.Theme.NewClickable(false),
+		isDexFirstVisit:   true,
 	}
+
+	dp.initSplashPageWidgets()
+	dp.finalizeOnboardingBtn = dp.Theme.Button(values.String(values.StrStartTrading))
```
[the remark this answers is not in the document]
Comment (ukane-philemon) [reply ui/page/dcrdex/dcrdex_page.go]: resolved.

**Review by dreacot — APPROVED** [review]

**Pull request merged**